api/src/main/java/org/pmiops/workbench/access/AccessModuleServiceImpl.java

```diff
@@ -0,0 +1,139 @@
+
+@Service
+public class AccessModuleServiceImpl implements AccessModuleService {
+  private static final Logger logger = Logger.getLogger(AccessModuleServiceImpl.class.getName());
+
+  private final Provider<List<DbAccessModule>> dbAccessModulesProvider;
+  private final Clock clock;
+
+  private final UserAccessModuleDao userAccessModuleDao;
+  private final UserDao userDao;
+  private final UserServiceAuditor userServiceAuditor;
+  private final Provider<WorkbenchConfig> configProvider;
+
+  private static final BiMap<AccessModule, AccessModuleName> CLIENT_TO_STORAGE_ACCESS_MODULE =
+      ImmutableBiMap.<AccessModule, AccessModuleName>builder()
+          .put(AccessModule.TWO_FACTOR_AUTH, AccessModuleName.TWO_FACTOR_AUTH)
+          .put(AccessModule.ERA_COMMONS, AccessModuleName.ERA_COMMONS)
+          .put(AccessModule.COMPLIANCE_TRAINING, AccessModuleName.RT_COMPLIANCE_TRAINING)
+          .put(AccessModule.RAS_LINK_LOGIN_GOV, AccessModuleName.RAS_LOGIN_GOV)
+          .put(AccessModule.DATA_USE_AGREEMENT, AccessModuleName.DATA_USER_CODE_OF_CONDUCT)
+          .put(AccessModule.PUBLICATION_CONFIRMATION, AccessModuleName.PUBLICATION_CONFIRMATION)
+          .put(AccessModule.PROFILE_CONFIRMATION, AccessModuleName.PROFILE_CONFIRMATION)
+          .build();
+
+  private static final BiMap<BypassTimeTargetProperty, AccessModuleName>
+      AUDIT_TO_STORAGE_ACCESS_MODULE =
+          ImmutableBiMap.<BypassTimeTargetProperty, AccessModuleName>builder()
+              .put(BypassTimeTargetProperty.ERA_COMMONS_BYPASS_TIME, AccessModuleName.ERA_COMMONS)
+              .put(
+                  BypassTimeTargetProperty.COMPLIANCE_TRAINING_BYPASS_TIME,
+                  AccessModuleName.RT_COMPLIANCE_TRAINING)
+              .put(
+                  BypassTimeTargetProperty.TWO_FACTOR_AUTH_BYPASS_TIME,
+                  AccessModuleName.TWO_FACTOR_AUTH)
+              .put(BypassTimeTargetProperty.RAS_LINK_LOGIN_GOV, AccessModuleName.RAS_LOGIN_GOV)
+              .put(
+                  BypassTimeTargetProperty.DATA_USE_AGREEMENT_BYPASS_TIME,
+                  AccessModuleName.DATA_USER_CODE_OF_CONDUCT)
+              .build();
+
+  @Autowired
+  public AccessModuleServiceImpl(
+      Provider<List<DbAccessModule>> dbAccessModulesProvider,
+      Clock clock,
+      UserAccessModuleDao userAccessModuleDao,
+      UserDao userDao,
+      UserServiceAuditor userServiceAuditor,
+      Provider<WorkbenchConfig> configProvider) {
+    this.dbAccessModulesProvider = dbAccessModulesProvider;
+    this.clock = clock;
+    this.userAccessModuleDao = userAccessModuleDao;
+    this.userDao = userDao;
+    this.userServiceAuditor = userServiceAuditor;
+    this.configProvider = configProvider;
+  }
+
+  @Override
+  public void updateBypassTime(long userId, AccessBypassRequest accessBypassRequest) {
+    final DbUser user = userDao.findUserByUserId(userId);
+    final List<DbUserAccessModule> dbUserAccessModules = userAccessModuleDao.getAllByUser(user);
```

Comment:
Instead of load by user id and access module id, I load all dbUserAccessModules, then filter the one that is changing. 
The reason is the current table index is on user_id only, will loading user_id and module id has a worse performance? 
And a user may just have ~7 modules at most, so it would be fine to load them all then do the filter. 
But I am not familiar with mysql index, if load by user_id + module_id have the same performance, I can switch to that.

Reply:
I don't know which would be better - either seems fine as a first implementation